USER_GUIDE.md

```diff
@@ -368,4 +368,51 @@ The plugin defaults to `appengine-web.xml` based deployment if your project cont
 file. If your project also has an `src/main/appengine/app.yaml` file and you wish to use that, you may temporarily move the
 `appengine-web.xml` file to a different location before deploying.
 
+### How do I deploy a thin jar to App Engine?
+
+You can do this if you have an `app.yaml` based project and you have a custom `entrypoint` configured to start it.
+
+You will need to configure the stage goal to include your project's dependencies. You could use something like the
+[`maven-dependency-plugin:copy-dependency`](https://maven.apache.org/plugins/maven-dependency-plugin/copy-dependencies-mojo.html)
+goal to copy dependencies into a directory and then configure that directory to be staged with your application using
+the `extraFilesDirectories` parameter in the appengine plugin. For example:
+
+```xml
+<plugin>
+  <artifactId>maven-dependency-plugin</artifactId>
+  <executions>
+    <execution>
+      <phase>package</phase>
+      <goals>
+        <goal>copy-dependencies</goal>
+      </goals>
+      <configuration>
+        <outputDirectory>${project.build.directory}/dependencies</outputDirectory>
+      </configuration>
+    </execution>
+  </executions>
+</plugin>
+<plugin>
+  <groupId>com.google.cloud.tools</groupId>
+  <artifactId>appengine-maven-plugin</artifactId>
+  <version>2.1.0</version>
+  <configuration>
+    <!-- your config -->
+    <extraFilesDirectories>
+      <extraFilesDirectory>${project.build.directory}/dependencies</extraFilesDirectory>
+    </extraFilesDirectories>
+  </configuration>
+</plugin>
+```
+
+Then when you run `mvn package appengine:deploy` the dependencies are copied and included as part of the deployment.
```

Comment:
I could be wrong about this, but just in case:

I think a lot of frameworks nowadays will by default create a fat JAR, so maybe it'd be nice to explain a bit for those cases to deploy an app correctly. For example, Spring Boot will generate two JARs (fat one and the original one), and isn't it that this command will copy the fat JAR to the staging area (or both)?

BTW, there seems to be a working hack the user and @ludoch used to remove a fat JAR.

Reply:
Yes, I think I might lean towards thinking that that's on them. For springboot, in maven, I believe some of the configurations leads to it overriding the default final artifact as the springboot fat jar (but I could be wrong).